cli/pkg/state/vmclarity.go

```diff
@@ -40,29 +40,36 @@ type VMClarityState struct {
 	scanResultID models.ScanResultID
 }
 
+// nolint:cyclop
 func (v *VMClarityState) WaitForVolumeAttachment(ctx context.Context) error {
-	ctx, cancel := context.WithTimeout(ctx, DefaultWaitForVolTimeout)
-	defer cancel()
+	logger := log.GetLoggerFromContextOrDiscard(ctx)
 
-	timer := time.NewTimer(DefaultWaitForVolRetryInterval)
+	timer := time.NewTicker(DefaultWaitForVolRetryInterval)
 	defer timer.Stop()
 
 	for {
 		select {
 		case <-timer.C:
 			status, err := v.client.GetScanResultStatus(ctx, v.scanResultID)
 			if err != nil {
-				return fmt.Errorf("failed to get scan result status: %w", err)
+				logger.Errorf("failed to get scan result status: %v", err)
```

Comment:
I wanted to make sure that the `scanner` won't fail in case there is a transient error where the `scanner` cannot reach the `backend` API. e.g. VMClarity server is being upgraded therefore it is not available for short period of time.

Reply:
it'll retry on the next tick anyway until its aborted by the context so I think logging is ok here, WDYT?

Reply:
> So maybe a retry logic is better, it feels that if the scanner will just skip the error here it will cause unexpected behavior later.

There is already an implicit retry logic here as the `break` only returns from the `select` but not from the `for` loop. So it will retry when the next arrives on `timer.C` channel.

Reply:
cool, missed it. tnx.